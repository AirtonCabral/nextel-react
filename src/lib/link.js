import React from 'react'
import { Link } from 'react-router-dom'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { loadPage } from '../actions/a_dom'


export class ConnectedLink extends React.Component {
    handleClick = () => {
        if (this.props.loadPage) this.props.loadPage(this.props.to)
        else console.warn('ConnectedLink foi chamado sem especificar prop loadPage')
    }

    render() {
        // Evita warning de DOM
        const filtered = { ...this.props }
        delete filtered.loadPage

        return (
            <Link {...filtered} onClick={this.handleClick}>{this.props.children}</Link>
        )
    }
}

const mapDispatchToProps = dispatch => bindActionCreators({
    loadPage
}, dispatch)

export default connect(() => ({}), mapDispatchToProps)(ConnectedLink)