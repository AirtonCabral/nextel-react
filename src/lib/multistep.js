import React from 'react'

class MultiStep extends React.Component {
    state = {
        defaultStep: 0
    }

    nextStep = () => {
        this.setState({defaultStep: this.state.defaultStep + 1})
    }

    prevStep = () => {
        this.setState({defaultStep: this.state.defaultStep - 1})
    }

    renderComponents = () => {
        return this.props.steps.map((item, index) => {
            if(this.state.defaultStep == index)
                return <React.Fragment key={index}>{item.component}</React.Fragment>
        })
    }
    
    renderHeader = () => {
        return(
            <div style={{display: 'flex', flexDirection: 'column', justifyContent: 'space-evenly', textAlign: 'center', padding: 20, backgroundColor: '#281c6a', height: 170}}>
                <h2 style={{color: '#fff', marginTop: 0, fontWeight: 400}}>{this.props.titleHeader}</h2>
                <div style={{display: 'flex', justifyContent: 'space-evenly'}}>
                <div style={{top: 116, position: 'absolute', height: 1, backgroundColor: '#22ebc0', width: '50%'}}></div>
                {
                    this.props.steps.map((item, index) => {
                        return(
                            <div key={index} style={{color: '#fff', zIndex: 1}}>
                                <div>{item.title}</div>
                                <div style={{width: 45, height: 45, display: 'flex', justifyContent: 'center', alignItems: 'center', borderRadius: 50, backgroundColor: '#fff', color: '#22ebc0', borderColor: '#22ebc0', margin: '10px auto'}}>{index + 1}</div>
                            </div>
                        )
                    })
                }
                </div>
            </div>
        )
    }
    
    render() {
        return(
            <React.Fragment>
                {this.renderHeader()}
                {this.renderComponents()}
            </React.Fragment>
        )
    }
}

export default MultiStep