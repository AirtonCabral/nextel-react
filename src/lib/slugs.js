
export function toSlug(string) {
    let slug = string.trim()
    slug = slug.toLowerCase()

    const from = 'àáãäâèéëêìíïîòóöôùúüûñç·_,:;/'
    const to =   'aaaaaeeeeiiiioooouuuunc-----_'

    for (let i = 0, l = from.length; i < l; i += 1) {
        slug = slug.replace(new RegExp(from.charAt(i), 'g'), to.charAt(i))
    }

    slug = slug.replace(/[^a-z0-9 _-]/g, '-') // Troca caracteres inválidos por -
        .replace(/\s+/g, '-') // Troca espaço em branco por -
        .replace(/-+/g, '-') // Transforma múltiplos "-" por um
        .replace(/^[^a-z0-9]/, '') // Remova primeiro caracter que não seja alfanumérico
        .replace(/[^a-z0-9]$/, '') // Remova última caracter que não seja alfanumérico

    return slug
}
