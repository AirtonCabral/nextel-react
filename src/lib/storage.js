import CircularJSON from 'circular-json'

export const loadState = () => {
    try {
        const serializedState = localStorage.getItem('state')
        if (serializedState === null) {
            return undefined
        }

        return CircularJSON.parse(serializedState)
    } catch (error) {
        return undefined
    }
}

export const saveState = (state) => {
    try {
        const serializedState = CircularJSON.stringify(state)
        localStorage.setItem('state', serializedState)
    } catch (error) {
        console.error(error)
    }
}

export const loadStateFromSession = () => {
    try {
        const serializedState = sessionStorage.getItem('state')
        if (serializedState === null) {
            return undefined
        }

        return CircularJSON.parse(serializedState)
    } catch (error) {
        return undefined
    }
}

 export const saveStateToSession = (state) => {
    try {
        const serializedState = CircularJSON.stringify(state)
        sessionStorage.setItem('state', serializedState)
    } catch (error) {
        console.error(error)
    }
}


export const loadObjectFromSession = (name) => {
    try {
        const serializedState = sessionStorage.getItem(name)
        if (serializedState === null) {
            return undefined
        }

        return CircularJSON.parse(serializedState)
    } catch (error) {
        return undefined
    }
}

export const saveObjectToSession = (object, name) => {
    try {
        const serializeObject = CircularJSON.stringify(object)
        sessionStorage.setItem(name, serializeObject)
    } catch (error) {
        console.error(error)
    }
}