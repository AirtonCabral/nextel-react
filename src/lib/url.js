export function parseGetParams(location) {
    let params = {}
    let query = location.search.substring(1)
    
    if (query.length === 0) return {}

    query.split('&').forEach((filter) => { 
        let item = filter.split('=')
        params[item[0]] = decodeURIComponent(item[1])
    })

    return params
}