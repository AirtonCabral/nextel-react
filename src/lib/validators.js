import store, { history } from '../store'

// Validação de Email
export function email_isValid(email) {
    return email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)
}

export function loginRequired() {
    if (store.getState().auth.user == null) {
        history.push('/login')
    }
}
