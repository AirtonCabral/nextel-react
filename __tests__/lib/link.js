import React from 'react'
import shallow from 'react-test-renderer/shallow'
import { Router } from 'react-router-dom'
import { ConnectedLink } from '../../src/lib/link'
import configureMockStore from 'redux-mock-store';

// Necessário usar renderer shallow (apenas 1 camada), para evitar que o react reclame que o 
// Link precisa de um pai router, o que complicaria muito o teste.
const renderer = new shallow()
const mockStore = configureMockStore();

describe('ConnectedLink', () => {
    it('Renderiza corretamente', () => {
        const tree = renderer.render(<ConnectedLink to="/home"/>)
        expect(tree).toMatchSnapshot()
    })

    it('Chama callback loadPage no onClick com argumentos corretos', () => {

        // loadPage mocked. Em uso real, não precisa colocar a função nos props, pois o redux cuida disso.
        const loadPage = jest.fn().mockImplementation(() => {})
        const link = renderer.render(<ConnectedLink  loadPage={loadPage} to="/home">Link</ConnectedLink>)

        // Chama o onClick, e verifica se loadPage foi chamado com o argumento correto.
        link.props.onClick()
        expect(loadPage).toBeCalledWith('/home')
    })

    // No último teste falei que o redux cuida disso, mas e se isso não acontecer? 
    const warn = jest.spyOn(console, 'warn')
    warn.mockImplementation(() => {})

    it('Reclama com warning se loadPage não for passado', () => {
        const link = renderer.render(<ConnectedLink to="/home">Link</ConnectedLink>)
        link.props.onClick()

        expect(warn).toBeCalledWith('ConnectedLink foi chamado sem especificar prop loadPage')
    })
})