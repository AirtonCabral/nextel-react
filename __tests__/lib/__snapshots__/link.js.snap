// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedLink Renderiza corretamente 1`] = `
<Link
  onClick={[Function]}
  replace={false}
  to="/home"
/>
`;
